tech-specs/src/specs/ledger.md

```diff
@@ -0,0 +1,190 @@
+# The ledger
+
+## The protocol
+
+- TODO describe DKG transactions
+
+### Transactions
+
+A transaction [encoded with proto3](./encoding.md#transactions) received from ABCI `DeliverTx` method is executed in two main steps:
+
+1. [Transaction execution](#transaction-execution)
+1. [Validity predicates check](#validity-predicates-check)
+
+#### Transaction execution
+
+For any error encountered in any of the following steps of transaction execution, the protocol MUST charge the gas used by the transaction and discard any storage changes that the transaction attempted to perform.
+
+1. Charge a base transaction [gas](#gas): \\[ \verb|BASE_TRANSACTION_FEE| \\]
+1. Decode the transaction bytes and validate the data.
```

Comment:
What kind of data validation is performed?

Reply:
Currently, we only require that the `timestamp` is specified. Updated